packages/remark-captions/__tests__/index.js

```diff
@@ -281,3 +290,75 @@ test('legend in paragraph', () => {
   `.replace(/·/g, ' '))
   expect(contents).toMatchSnapshot()
 })
+
+
+test('compiles to markdown', () => {
+  const md = dedent`
+    foo
+
+    ![](img)
+    Figure: 1 this is parsed as legend
+
+    baz
+
+    ![](img)
+    aFigure: 2 this is displayed as text
+
+    ![alt 2b](https://zestedesavoir.com/static/images/home-clem.4a2f792744c9.png)
+    this is displayed as text
+
+    foo
+    ![](img)
+    Figure: 3 this is a legend
+
+    ![](img)
+    Figure: 4 this is a legend, remainder of the paragraph goes into
+    the
+    legend
+
+    Figure: 5 this is a text with and image
+    ![](<title> img)
+    in the middle
+
+    Figure: 6 displayed as text
+    ![](img)
+    Figure: 7 is a legend
+
+    ![](img)
+    Figure: 8 is a legend.
+
+    Figure: 9 this is a text.
+    
+    > My citation
+    Source: first capt
+    Source: last capt··
+    2nd line
+
+    noop
+
+    > foo
+    > bar
+    > baz
+    > qux
+    Source: **first**
+    b*a*r
+    Source: This is **the real** \`source\`
+
+    noop
+    
+    > foo
+    > bar
+    > baz
+    > qux
+    Source: **first**
+    b*a*r
+    Source: This is **the real** \`source\`
+  `
```

Comment:
Please split these into a few separate cases. All the input syntax should be supported, for instance

    ```python
    print('bla')
    ```
    Code: figcapt1

returns `undefined figcapt1`